File: modules/search/examples/autocomplete-sample.py
from couchbase.cluster import Cluster, ClusterOptions
from couchbase.auth import PasswordAuthenticator
from couchbase.exceptions import CouchbaseException
import couchbase.search as search
import os

username = os.getenv("$CB_USERNAME$", default=None)
password = os.getenv("$CB_PASSWORD$", default=None)
hostname = os.getenv("$CB_HOSTNAME$", default=None)

cluster = Cluster.connect(
    "couchbase://" + hostname,
    ClusterOptions(PasswordAuthenticator(username,password)))

try:
    inputStr = "Beaufort Hotel"
    for i in range(2, min(8,len(inputStr))):
        testStr = inputStr[0:i]
        print("Input <" + testStr + ">, len: " + str(len(testStr)));

        result = cluster.search_query(
            "e-ngram-2-8", 
            search.QueryStringQuery(testStr),
            search.SearchOptions(limit=8, fields=["name"]))

        for row in result.rows():
            print(row.id,row.fields)

except CouchbaseException as ex:
    import traceback
    traceback.print_exc()

File: modules/search/examples/autocomplete-sample.go
package main

import (
    "os"
    "fmt"
    "math"
    "log"
    "github.com/couchbase/gocb/v2"
    "github.com/couchbase/gocb/v2/search"
)

func main() {
    cluster, err := gocb.Connect(
	os.Getenv("CB_HOSTNAME"),                    
	gocb.ClusterOptions{
        Authenticator: gocb.PasswordAuthenticator{
            Username: os.Getenv("CB_USERNAME"),      
            Password: os.Getenv("CB_PASSWORD"),      
        },
    })
    if err != nil {
        log.Fatal(err)
    }

    iterStr := "Beaufort Hotel"
    maxsz := int(math.Min(float64(len(iterStr)), float64(8)))
    for i := 2;  i <= maxsz; i++ {
        testStr := iterStr[0:i]
	fmt.Printf("Input <%s>, len: %d\n", testStr, len(testStr));
	results, err := cluster.SearchQuery(
	    "e-ngram-2-8",
	    search.NewQueryStringQuery(testStr), 
	    &gocb.SearchOptions { Fields: []string{"name"}, Limit: 8, },
	)
	if err != nil {
	    log.Fatal(err)
	}

	for results.Next() {
	    row := results.Row()
	    docID := row.ID
	    var fields interface {}
	    err := row.Fields( & fields)
	    if err != nil {
		panic(err)
	    }
	    fmt.Printf("Document ID: %s, fields: %v\n", docID, fields)
	}

	err = results.Err()
	if err != nil {
	    log.Fatal(err)
	}
    }
}
